contracts/integrations/test/exchange/transaction_protocol_fee_test.ts

```diff
@@ -0,0 +1,420 @@
+
+    before(async () => {
+        deployment = await DeploymentManager.deployAsync(env, {
+            numErc20TokensToDeploy: 4,
+            numErc721TokensToDeploy: 0,
+            numErc1155TokensToDeploy: 0,
+        });
+        const assetDataEncoder = new IAssetDataContract(constants.NULL_ADDRESS, env.provider);
+        const [makerToken, takerToken, makerFeeToken, takerFeeToken] = deployment.tokens.erc20;
+
+        alice = new Taker({ name: 'Alice', deployment });
+        bob = new Taker({ name: 'Bob', deployment });
+        charlie = new Taker({ name: 'Charlie', deployment });
+        feeRecipient = new FeeRecipient({
+            name: 'Fee recipient',
+            deployment,
+        });
+        maker = new Maker({
+            name: 'Maker',
+            deployment,
+            orderConfig: {
+                feeRecipientAddress: feeRecipient.address,
+                makerAssetData: assetDataEncoder.ERC20Token(makerToken.address).getABIEncodedTransactionData(),
+                takerAssetData: assetDataEncoder.ERC20Token(takerToken.address).getABIEncodedTransactionData(),
+                makerFeeAssetData: assetDataEncoder.ERC20Token(makerFeeToken.address).getABIEncodedTransactionData(),
+                takerFeeAssetData: assetDataEncoder.ERC20Token(takerFeeToken.address).getABIEncodedTransactionData(),
+            },
+        });
+
+        for (const taker of [alice, bob, charlie]) {
+            await taker.configureERC20TokenAsync(takerToken);
+            await taker.configureERC20TokenAsync(takerFeeToken);
+            await taker.configureERC20TokenAsync(deployment.tokens.weth, deployment.staking.stakingProxy.address);
+        }
+        await maker.configureERC20TokenAsync(makerToken);
+        await maker.configureERC20TokenAsync(makerFeeToken);
+
+        balanceStore = new BlockchainBalanceStore(
+            {
+                ...actorAddressesByName([alice, bob, charlie, maker, feeRecipient]),
+                StakingProxy: deployment.staking.stakingProxy.address,
+            },
+            { erc20: { makerToken, takerToken, makerFeeToken, takerFeeToken, wETH: deployment.tokens.weth } },
+            {},
+        );
+        await balanceStore.updateBalancesAsync();
+
+        order = await maker.signOrderAsync();
+        let data = exchangeDataEncoder.encodeOrdersToExchangeData(ExchangeFunctionName.FillOrder, [order]);
+        transactionA = await alice.signTransactionAsync({ data });
+
+        order = await maker.signOrderAsync();
+        data = exchangeDataEncoder.encodeOrdersToExchangeData(ExchangeFunctionName.FillOrder, [order]);
+        transactionB = await bob.signTransactionAsync({ data });
+
+        order = await maker.signOrderAsync();
+        data = exchangeDataEncoder.encodeOrdersToExchangeData(ExchangeFunctionName.FillOrder, [order]);
+        transactionC = await charlie.signTransactionAsync({ data });
+    });
+
+    after(async () => {
+        Actor.count = 0;
```

Comment:
Is this mandatory in every `after` smt now? I wonder if there's an abstraction that could more explicitly communicate what we're doing here. Ex, `Actor.reset()` or something.